client/gutenberg/editor/calypsoify-iframe.jsx

```diff
@@ -123,10 +131,22 @@ class CalypsoifyIframe extends Component {
 			const formattedMedia = map( media.items, item => mediaCalypsoToGutenberg( item ) );
 			const payload = multiple ? formattedMedia : formattedMedia[ 0 ];
 
-			this.iframePort.postMessage( {
-				action: 'selectMedia',
-				payload,
-			} );
+			if ( this.mediaSelectPort ) {
+				this.mediaSelectPort.postMessage( payload );
+
+				// this is a once-only port
+				// after sending our message we want to close it out
+				// and prevent sending more messages (which will be ignored)
+				this.mediaSelectPort.close();
+				this.mediaSelectPort = null;
```

Comment:
boo

> Authors are strongly encouraged to explicitly close MessagePort objects to disentangle them, so that their resources can be recollected. Creating many MessagePort objects and discarding them without closing them can lead to high transient memory usage since garbage collection is not necessarily performed promptly, especially for MessagePorts where garbage collection can involve cross-process coordination.

https://html.spec.whatwg.org/multipage/web-messaging.html#ports-and-garbage-collection